packages/core/src/ceramic.ts

```diff
@@ -455,7 +459,19 @@ export class Ceramic implements CeramicApi {
       }
 
       if (!this._gateway) {
-        await this.anchorService.init()
+        const carBuilder = this.repository._internals.anchorRequestCarBuilder
+        const repository = this.repository
+        const handler: AnchorLoopHandler = {
+          buildRequestCar(streamId: StreamID, tip: CID): Promise<CAR> {
+            return carBuilder.build(streamId, tip)
+          },
+          async handle(event: AnchorEvent): Promise<boolean> {
+            const state$ = await repository.fromMemoryOrStore(event.streamId)
+            if (!state$) return true
+            return repository._internals.handleAnchorResponse2(state$, event)
```

Comment:
this is introducing another circular dependency, just as I'm trying to hunt down and remove the last of these from the Repository.  Instead of the AnchorService calling back into the Repository as it gets anchor responses, can there just be an Observable that spits out anchor commits in need of handling, and the Repository can subscribe to that observable and handle the anchor commits as they come out?

As in, basically similar to how it works today, but instead of one Observable per stream there is one global Observable of all streams

Reply:
Oh, the circular dependencies gave me so much pain! There might be an option to clean that up, but for this particular case, I see no other way. Maybe after all the other stuff from here gets unbundled.

Observables are not an option. We need to know on the loop side, that an event is terminal or not. There is no sane way one could signal that back to the loop. Streams (NodeJS and Web Streams alike) go the same route for the same reason. Iterables/Generators maybe, yet it's a fucking joke to control their lifecycle.